teuthology/dispatcher/supervisor.py

```diff
@@ -151,6 +153,45 @@ def run_job(job_config, teuth_bin_path, archive_dir, verbose):
     if 'targets' in job_config:
         unlock_targets(job_config)
 
+def mark_machine_down(machine_name, count):
+    teuthology.lock.ops.update_lock(machine_name,
+                                    description='reimage failed {0} times'.format(count),
+                                    status='down',
+                                    )
+    log.exception(
+        'Reimage failed {0} times ... marking machine down'.format(count)
+    )
+
+def failure_is_reimage(failure_reason):
+    if not failure_reason:
+        return False
+    reimage_failure = "Error reimaging machines:"
+    if reimage_failure in failure_reason:
+        return True
+    else:
+        return False
+
+def check_for_failures_and_mark_down(targets, count=10):
```

Comment:
Even though it's really long, I think `check_for_reimage_failures_and_mark_down` is a little better